internal/appsec/waf.go

```diff
@@ -109,10 +110,35 @@ func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout tim
 			return
 		}
 
+		// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
+		// see if the associated user should be blocked. Since we don't control the execution flow in this case
+		// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
+		op.On(sharedsec.OnUserIDOperationStart(func(operation *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
+			values := map[string]interface{}{}
+			for _, addr := range addresses {
+				if addr == userIDAddr {
+					values[userIDAddr] = args.UserID
+				}
+			}
```

Comment:
Sounds good, it will really clean things up to get rid of those weird loops. I'll track this [here](https://datadoghq.atlassian.net/browse/APPSEC-8157) and bump the prio to medium.